src/ts/options/index.ts

```diff
@@ -30,8 +39,39 @@ const logOut = async () => {
 	setClass(true, false);
 };
 
+const setTagIndex = async () => (tagIndexInput().value = (await config.get(ConfigKey.SpreadsheetLink)) ?? "");
+
+const onValidValue = (valid: (value: string) => void, invalid?: () => void) => async () => {
+	const {value} = tagIndexInput();
+	if (await isValidSheetLink(value)) {
+		return valid(value);
+	} else {
+		return invalid?.();
+	}
+};
+
+const getTagIndex = onValidValue(async (value) => {
+	await config.set(ConfigKey.SpreadsheetLink, value);
+	moreInfo().innerText = "SAVED";
+	saveButton().disabled = true;
+	return invokeWorker(WorkerKind.DispatchEvent, BackgroundEvent.AddedSheetLink);
+});
```

Comment:
why no invalid callback for this

Reply:
OH yeah I didn't wanna think about what the page should do when you try to save an invalid value, so I just hide the button entirely if there is an invalid value